src/components/data-display/charts/RNPChart.tsx

```diff
@@ -180,9 +183,20 @@ export function RNPChart({
     return <></>;
   };
 
-  if (!chartData.length) return <></>;
+  if (!chartData?.length)
+    return (
+      <span className="h-fit flex flex-col text-white w-full items-center p-5 justify-center m-auto">
+        <Lottie animationData={arioLoading} loop={true} className="h-[100px]" />
+        <span>Loading Price Chart...</span>
+      </span>
+    );
 
-  if (!costDetails?.returnedNameDetails) return <></>;
+  if (!costDetails?.returnedNameDetails)
+    return (
+      <div className="flex text-white text-lg justify-center items-center m-auto">
+        This is not a returned name, unable to load chart.
```

Comment:
Odd, seems like if this is the case, this shouldn't really be shown and error handled elsewhere. Could return empty <></> for example here.

Reply:
Yeah in theory this should never be reached 🤔 ok will just do a fragment